### PR TITLE
added redux folder

### PR DESCRIPTION
**Pull request by Booze33**
<h1>Firstly</h1>

- Add Redux Toolkit 
- Create redux folder
- Configure a Redux store

<h1>Second</h1>

- Define a slice of state for books and categories
- Add initialState and reducers

<h1>Third</h1>

- Align the structure of files.

### CODE REVIEW

File: src/redux/category/categorySplice.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  categories: 'Under Construction',
+};
+
+const categoriesSlice = createSlice({
+  name: 'Categories',
+  initialState,
+  reducers: {},
+});
+
+export default categoriesSlice.reducers;
```
Comment (Shedrack-Sunday) [inline src/redux/category/categorySplice.js]: - [ ] Good job you did her 👏 . However based on project requirements, you're to define an action type for checking the status. Then export the `checkStatus` as an action

## Possible implementation:
```
reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
```
## Requirement 

image

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Booze33       👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well so far in the project,  However, there are a few you need to work on, to move to the next project. 💯

 


## Highlights   
- Good use of GitHub features ✔️   
- Nice use of Redux 👍 
- Clean component usage ✔️ 

  
# Required Changes  🌵 

- [x] You made some impressive reviews in the comments, which I must say is a job well done. However, there's one you missed 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/redux/category/categorySplice.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  reducers: {
+    checkStatus: (state, action) => ({
+      ...state,
+      categories: action.payload === 'Under construction'
+        ? 'Under construction'
+        : state.categories,
+    }),
+  },
+});
+
```
Comment (brainconnect93) [inline src/redux/category/categorySplice.js:19]: - [ ] As requested by the previous reviewer, it is also part of the project requirement, kindly export the `checkStatus` as an action. With this you would be sure you have follow the project requirement strictly and implemented the changes requested by the previous reviewer.

Possible implementation :point_down:

```JS
export const { checkStatus } = categoriesSlice.actions
```

**Review by brainconnect93 — CHANGES_REQUESTED** [review]
### Hi @Booze33, 👋 

# STATUS: Changes Required :recycle:

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To Highlight :1st_place_medal: 

- Descriptive Pull Request. :heavy_check_mark:
- Action and Reducer functionality. :heavy_check_mark:
- Ducks pattern file structure. :heavy_check_mark:
- `@reduxjs/toolkit` & `react-redux` present in package.json :heavy_check_mark:
- Linters environment properly setup :heavy_check_mark:
- Good use of Git flow :heavy_check_mark: 

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @brainconnect93 ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bobb-Rob — APPROVED** [review]
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @Booze33 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ No Linter error
✅ Correct use of Git flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**